event.go

```diff
@@ -17,337 +17,329 @@ package gomatrixserverlib
 
 import (
 	"bytes"
+	"crypto/ed25519"
 	"encoding/json"
-	"errors"
 	"fmt"
-	"reflect"
 	"strings"
-	"unsafe"
+	"time"
 
 	"github.com/matrix-org/gomatrixserverlib/spec"
+	"github.com/matrix-org/util"
 	"github.com/tidwall/gjson"
 	"github.com/tidwall/sjson"
-	"golang.org/x/crypto/ed25519"
 )
 
-// Event validation errors
-const (
-	EventValidationTooLarge int = 1
-)
+type ProtoEvent struct {
```

Comment:
I don't understand the ratinale for replacing `event` with `ProtoEvent`. The former is a concrete implementation of a PDU, the latter is the JSON blob returned by `/make_join` and other proto events (e.g the initial set of events in `/createRoom`). Why are we conflating the two ideas?